packages/core/src/utils/Utils.ts

```diff
@@ -183,6 +186,38 @@ export function parseJsonSafe<T = unknown>(value: unknown): T {
   return value as T;
 }
 
+function getEntityManager(caller: { orm?: MikroORM; em?: EntityManager }, context: unknown): EntityManager | undefined {
+  if (context instanceof EntityManager) {
+    return context;
+  }
+
+  if (context instanceof EntityRepository) {
+    return context.getEntityManager();
+  }
+
+  if (context instanceof MikroORM) {
+    return context.em;
+  }
+
+  if (caller.em instanceof EntityManager) {
+    return caller.em;
+  }
+
+  if (caller.orm instanceof MikroORM) {
+    return caller.orm.em;
+  }
+
+  return undefined;
+}
+
+/**
+ * Find entityManager in injected context, or else in class's orm or em properties.
+ */
+export async function resolveGetContext<T>(caller: T & { orm?: MaybePromise<MikroORM>; em?: EntityManager }, getContext?: GetContext<T>): Promise<EntityManager | undefined> {
+  const context = typeof getContext === 'function' ? await getContext(caller) : await getContext;
+  return getEntityManager({ orm: await caller.orm, em: caller.em }, context);
```

Comment:
the EM can be also async callback
```suggestion
  return getEntityManager({ orm: await caller.orm, em: await caller.em }, context);
```